CMakeLists.txt

```diff
@@ -467,9 +467,9 @@ endif (WITH_RADOSGW)
 option(WITH_CEPHFS "CephFS is enabled" ON)
 
 if(NOT WIN32)
-# Please specify 3.[0-7] if you want to build with a certain version of python3.
-set(WITH_PYTHON3 "3" CACHE STRING "build with specified python3 version")
-if(NOT WITH_PYTHON3 STREQUAL "3")
+# Please specify 3.[0-8] if you want to build with a certain version of python3.
+set(WITH_PYTHON3 "3.8" CACHE STRING "build with specified python3 version")
```

Comment:
I don't have 3.8 installed on my system. Is there a way to make exclude 3.5 without mandating versions that are not installed?

Reply:
i don't think wiring the default value to 3.8 helps unless we depends on 3.8 and up.

Reply:
> I don't have 3.8 installed on my system. Is there a way to make exclude 3.5 without mandating versions that are not installed?

@sebastian-philipp but if we declare the 3.8 dep it'll become installed. Is there any specific reason not to want 3.8 installed?

Reply:
> i don't think wiring the default value to 3.8 helps unless we depends on 3.8 and up.

Just to mention a single change: f-strings are not supported in 3.5, so we should at least indicate that we require 3.6 or above (which, on the other hand, is nothing really adventurous given that 3.5 end of life finished almost a year ago and 3.5 EOL is Dec this year).

The reason to go with 3.8 is using improved typing constructs and other basic helpers as `dataclasses`.

Reply:
> > i don't think wiring the default value to 3.8 helps unless we depends on 3.8 and up.
> 
> @tchaikov just to mention a single change: f-strings are not supported in 3.5, so we should at least indicate that we require 3.6 or above (which, on the other hand, is nothing really adventurous given that 3.5 end of life finished almost a year ago and 3.6 EOL is Dec this year).

i agreed that depending on 3.6 is the way to go. that's actually one of the reasons why i pushed the xenial to bionic change.

please note, my concern is not why 3.6 but why 3.8.

Reply:
Personally I've found a few things that are of interest in python 3.8, but they mainly revolve around `asyncio` improvements. On top of that, I find having type annotation forward references is incredibly useful. And, as @epuertat mentioned, 3.6 is EOL at end of this year, and moving on to python 3.8 makes sense just because of that, should there be no particular constraint.